ChangeLog

```diff
@@ -1,3 +1,143 @@
+	- Honor ftp_port worker-queues option
+	- Log early level-0/1 debugs() messages to cache_log
+	- Support reliable zeroing of sensitive buffers
+	- Do not overwrite caching bans
+	- Avoid level-3 "failed to find or read error text file" WARNINGs
+	- Do not blame cache_peer for 4xx CONNECT responses
+	- Improve cache_peer reporting and NetDB cache_peer search
+	- Mimic GET reforwarding decisions when our CONNECT fails
+	- Report (problems with) DEAD-at-start cache_peers
+	- Do not lock an SMP hit twice for the same transaction
+	- Fewer "busy shutdown" crashes; simpler shutdown cleanup
+	- Do not tell admins that we are replacing their .* with our .*
+	- Discarded connections do not contribute to forward_max_tries
+	- Do not ApplyTcpKeepAlive() to PortCfg-unrelated traffic
+	- Honor assertions during shutdown
+	- Logformat %lp expands to "-" in wildcard listening port configs
+	- Kid restart leads to persistent queue overflows, delays/timeouts
+	- Do not stop listening after "ERROR: NAT/TPROXY lookup failed..."
+	- Do not skip problematic regexes in ACLs
+	- Improve coredump_dir on FreeBSD and Solaris based OS
+	- Avoid reverse DNS lookups when logformat %>A is unused
+	- BUG: Unexpected state while connecting to ... server
+	- Properly track (and mark) truncated store entries
+	- Support "file" syntax for 'squid_error' and 'has' ACL parameters
+	- Allow sending "squid -k ..." signals to PID 1
+	- Remove bogus "found KEY_PRIVATE" WARNINGs
+	- Add cache_log_message directive
+	- Avoid "BUG #3329: Lost orphan ..." during accept problems
+	- Add tls_key_log to report TLS communication secrets
+	- Do not duplicate free disk slots on diskers restart
+	- Report SMP store queues state (mgr:store_queues)
+	- Add paranoid_hit_validation directive
+	- Remove 8K limit for single access.log line
+	- Add %transport::>connection_id logformat code
+	- Rename ./configure option --with-libxml2 to --with-xml2
+	- Rename ./configure option --with-libcap to --with-cap
+	- Match ./configure --help parameter names with their defaults
+	- Fix typo in manager ACL
+	- Fix SQUID-MIB smilint errors
+	- Fix milliseconds in certain cache.log messages
+	- Fix ignore-cc/act-as-origin in wildcard split-stack ports
+	- Fix comm.cc:644: "address.port() != 0" assertion
+	- Fix StoreMap.cc "anchorAt(anchorId).reading()" assertions
+	- Fix double-free segmentation fault on shutdown
+	- Fix client_side_request.cc:2028 "request->method.id()" assertion
+	- Fix FATAL ServiceRep::putConnection exception: theBusyConns > 0
+	- Fix reconfiguration leaking tls-cert=... memory
+	- Fix X509_V_ERR_UNABLE_TO_GET_ISSUER_CERT_LOCALLY handling
+	- Fix "mem_obj->inmem_lo == 0" assertion in StoreEntry::swapOut()
+	- Fix TCP keepalive
+	- Fix SslBump reconfiguration leaking public key memory
+	- Fix ACL-related reconfiguration memory leak
+	- Fix socket accounting for TCP accept()
+	- ... and many documentation changes
+	- ... and much code cleanup and polishing
+	- ... and all fixes from 5.8
```

Comment:
IMO, v6 changes should be listed relative to some (disclosed!) v5 release. They should not require reading v5 change log. I suggest removing this line and (if we want to preserve the current everything-is-a-list-item format) starting the list with something like

```
	- The changes listed below are relative to Squid v5.8:
```

Reply:
No,
 1) the changelog is long enough already without having to list every change twice.

 2) persons reading a changelog containing a weave of parallel release series do need guidance of what changes are relevant to their version. This line is the mechanism by which that is provided.

 3) ChangeLog is read from backwards from the version of interest. This line documents that any intervening version(s) change lists are to be skipped; until the disclosed version is read. See the v5.0.x release sections of ChangeLog for how that works.

 4) the wording is kept consistent for shared interpretation between human and machine tooling.

Reply:
> the changelog is long enough already without having to list every change twice.

Please note that nobody is suggesting to list every change twice.

> persons reading a changelog containing a weave of parallel release series do need guidance of what changes are relevant to their version. 

Yes, and the suggested change will address that need better than the current PR code does.

> This line is the mechanism by which that is provided.

That fact does not address my change request: The change request is saying, in part, "we should not address a need for X by doing Y". A "we are doing Y" response can confirm the alleged problem, but cannot address the change request. Thus, I am not sure why you are stating this.

> This line documents that any intervening v(N-1) change lists are to be skipped; until the disclosed version is read. See the v5.0.x release sections of ChangeLog for how that works.

IMO, this approach does not work well, especially for a new major release. I am not disputing the fact that this approach is used or can be used. I am saying that we can (and should) do better.

> the wording is kept consistent for shared interpretation between human and machine tooling.

IMO, introducing improvements that are inconsistent with problematic past practices can be a step forward and now is a good moment to introduce those improvements and inconsistencies (among with format changes and other adjustments discussed [elsewhere](https://github.com/squid-cache/squid/pull/1245#pullrequestreview-1269568901)).

Reply:
Had a meeting with @kinkie today. We will leave this file as-is for now and do a full redesign for a later release soon.